### PR TITLE
adding more comments to why drops do not happen 

### PR DESCRIPTION
**Pull request by himanshug**
when there are no segments in db

### CODE REVIEW

**Comment by gianm** [comment]
I would prefer to have the rationale "done to prevent a merge condition…" in the docs and in the comment but not in the log message, just because it makes the log message really long. but I am ok with it either way so :+1:

**Comment by himanshug** [comment]
yeah that msg is verbose but provides full answer to why it happened or user will have to dig for more.

File: docs/content/design/coordinator.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Cleaning Up Segments
 --------------------
 
 Each run, the Druid coordinator compares the list of available database segments in the database with the current segments in the cluster. Segments that are not in the database but are still being served in the cluster are flagged and appended to a removal list. Segments that are overshadowed (their versions are too old and their data has been replaced by newer segments) are also dropped.
+Note that if all segments in database are deleted(or marked unused), then coordinator will not drop anything from the historicals. This is done to prevent a race condition in which the coordinator would drop all segments if it started running cleanup before it finished polling the database for available segments for the first time and believed that there were no segments.
```
Comment (fjy) [inline docs/content/design/coordinator.md]: A comment in the code is probably enough, but no harm repeating things I suppose

**Comment by fjy** [comment]
:+1:

**Pull request merged**